app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Storage;
+use App\Http\Requests\UpdateUserRequest;
+
+class UserController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $users = DB::table('users')
+            ->select('name', 'email', 'avatar')
+            ->orderBy('name', 'asc')
+            ->get();
+
+        return response()->json($users);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        try {
+            $user = DB::table('users')
+                ->select('name', 'email', 'avatar')
+                ->find($id);
+
+            return response()->json($user);
+        } catch (\Exception $e) {
+            return response()->json([
+                'message' => 'error',
+            ]);
+        }
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        try {
+            $user = DB::table('users')->find($id);
+
+            return response()->json($user);
+        } catch (\Exception $e) {
+            return response()->json([
+                'message' => 'error',
+            ]);
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(UpdateUserRequest $request, $id)
+    {
+        try {
+            $user = DB::table('users')->find($id);
+
+            if ($request->avatar == $user->avatar) {
+                $newNameAvatar = $request->avatar;
+            } else {
```

Comment:
consider removing this check, it's useless because there will be multi files with the same name